editor/debugger/script_editor_debugger.h

```diff
@@ -88,6 +88,11 @@ class ScriptEditorDebugger : public MarginContainer {
 	PopupMenu *item_menu;
 
 	EditorFileDialog *file_dialog;
+	enum FileDialogMode {
+		SAVE_MONITORS_CSV,
+		SAVE_VRAM_CSV,
+	};
+	FileDialogMode file_dialog_mode;
```

Comment:
The name is not so good, as `FileDialog` already has a `mode` property which is about save/load actions.

Reply:
The thing is that in 3.2 we already have multiple modes. Therefore, what I did is port that "modes infrastructure" (the enum, the variable and the `switch`) to 4.0 so I could add a new mode while keeping the code bases as convergent as possible. That would allow, among other things, to port the `SAVE_NODE` mode for 3.2 more easily.

This is how it looks in 3.2 currently: https://github.com/godotengine/godot/blob/69081c000dcc25a9534c8358d931af0b1b8b8dd2/editor/script_editor_debugger.h#L112

Reply:
@RandomShaper save node is already implemented in 4.0, implemented in the remote tree see https://github.com/godotengine/godot/blob/master/editor/debugger/editor_debugger_tree.cpp#L265

Reply:
Interesting... Then I'll do some renaming.

Reply:
Done!